tests/cpp/test_multidevice_lower_communication.cpp

```diff
@@ -56,11 +60,19 @@ using InOutMesh = std::pair<DeviceMesh, DeviceMesh>;
 
 static constexpr int kTensorSize = 4;
 
-class LowerGatherTest : public MultiDeviceTest,
-                        public testing::WithParamInterface<InOutMesh> {};
+class LowerGatherTest
+    : public MultiDeviceTest,
+      public testing::WithParamInterface<std::tuple<InOutMesh, bool>> {};
```

Comment:
```suggestion
      public testing::WithParamInterface<std::tuple<DeviceMesh, DeviceMesh, bool>> {};
```
I don't see a good reason for nesting. This will probably simplify the structured bindings below.

Reply:
I think this is here because we want certain pairs of DeviceMesh, and not every pair that Combine(InMesh, OutMesh) would give